clang/test/SemaCXX/matrix-types-pseudo-destructor.cpp

```diff
@@ -0,0 +1,31 @@
+// RUN: %clang_cc1 -fsyntax-only -fenable-matrix -std=c++11 -verify %s
+// expected-no-diagnostics
+
+template <typename T>
+void f() {
+  T().~T();
+}
+
+template <typename T>
+void f1(T *f) {
```

Comment:
Should we have a `constexpr` test to ensure that this noop works fine in a constant expression too?

Reply:
Probably a good idea

Reply:
Well, it turns out we currently don’t support matrix types in constant expressions, so, er, I’ll open a separate issue about that.

Reply:
#120167